### ISSUE TITLE
[indent] Ternary doesn't allow extended indentation

### ISSUE THREAD

**Issue by xixixao** [eslint/eslint#7698]
**Tell us about your environment**

* **ESLint Version:**
whatever is on the DEMO on eslint.org, unfortunately it doesn't show
* **Node Version:**
n/a
* **npm Version:**
m/a
* **What parser (default, Babel-ESLint, etc.) are you using?**
dunno, environments is set to `node`, `es6`

**Please show your full configuration:**
`indent` is on

**What did you do? Please include the actual source code causing the issue.**
```js
const foo = Math.random()
  ? {
      a: 3,
    }
  : {};
```

**What did you expect to happen?**
No warning.
**What actually happened? Please include the actual, raw output from ESLint.**
4:5 - Expected indentation of 2 spaces but found 4. (indent)

**Comment by xixixao** [comment]
@not-an-aardvark does your #7618 fix this too?

**Comment by not-an-aardvark** [comment]
No, this is (mostly) working as intended. The correct output would be:

```js
3:7 - Expected indentation of 4 spaces but found 6. (indent)
4:5 - Expected indentation of 2 spaces but found 4. (indent)
```

For what it's worth, I think this is a duplicate of https://github.com/eslint/eslint/issues/6606.

**Comment by not-an-aardvark** [comment]
To elaborate a bit more on why I think reporting an error here is correct behavior:

```js
const foo = Math.random()
  ? {
      a: 3,
    }
  : {};
```

The question mark is indented at two spaces, but the `a` is indented at 6 spaces. Presumably the location of the `a` is based on the column of the opening `{`, but the opening `{` isn't actually at the beginning of the line. So the `?` and the space after it are actually contributing to the indentation here, which isn't something that happens anywhere else in the `indent` rule.

To make this easier to see, suppose you decided to change your indent size to 4 spaces instead of 2:

```js
const foo = Math.random()
    ? {
        a: 3,
      }
    : {};
```

In this case the closing `}` is at an offset of 6 spaces. I think it would be very unexpected for the indent rule to enforce a 6-space offset when your config was trying to enforce indentation in 4-space levels.

**Comment by xixixao** [comment]
I agree this is duplicate of #6606. We use this style pretty consistently, including with JSX and other expressions, where lint doesn't error:
```es6
// works
const foo = Math.random()
  ? <div
      id="test",
    />
  : null;
// works
const foo = Math.random()
  ? parseInt(
      "42",
      16,
    )
  : 0;
```
I see that this doesn't work as well when the indentation isn't two spaces and don't really have an answer for that, as I personally don't have experience with JS using different config. Perhaps if this was a configuration flag, it would only be used by people who have indent set to 2.

**Comment by Janpot** [comment]
Could you consider the following for 4 space indentation?

```js
const foo = Math.random()
    ?   {
            a: 3,
        }
    :   {};
```

Or is that too far off?

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
